docs/proposals/healthchecks.md

```diff
@@ -0,0 +1,67 @@
+# HealthChecks
+
+
+- This will be container level, optional field, called `health`
+- Liveness and Readiness Probes will be generated out of this in the `deployment.spec.template.spec.containers`.
+
+## Sample example
+
+```yaml
+containers:
+- image: centos/httpd
+  health:
+  - ready: true
+    get: "http://localhost:8080/v1"
+    delay: 10
+    timeout: 3
+    tries: 5
+
+
+  - ready: true
+    exec: "cat /foo/bar"
+
+  - ready: true
+    get: "http://localhost:8080/healthz"
+
+  - live: true
+    exec: "cat /foo/bar"
+
+  - live: true
+    get: "http://localhost:8080/healthz"
+```
+
+## Field description
+
+| Field name | Type   | Optional | Conflicting field | Details                                                                                                       | Converted to (append to `deployment.spec.template.spec.containers`)                                                                          |
+|------------|--------|----------|-------------------|---------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `ready`    | bool   | no       | `live`            | Generate Readiness Probe config.                                                                              | `readinessProbe`                                                               |
+| `live`     | bool   | no       | `ready`           | Generate Liveness Probe config.                                                                               | `livenessProbe`                                                                |
+| `get`      | string | no       | `exec`            | A URL to do HTTP request on. Detailed information of what field in URL gets converted to what is given below. | `readinessProbe.httpGet` **OR** `livenessProbe.httpGet`                         |
+| `exec`     | string | no       | `get`             | A command to run inside the container.                                                                        | `readinessProbe.exec` **OR** `livenessProbe.exec`                               |
+| `delay`    | uint32 | yes      | None              | Number of seconds after the container has started before probes are initiated.                                | `livenessProbe.initialDelaySeconds` **OR** `readinessProbe.initialDelaySeconds` |
+| `timeout`  | uint32 | yes      | None              | Number of seconds after which the probe times out.                                                            | `livenessProbe.timeoutSeconds` **OR** `readinessProbe.timeoutSeconds`           |
+| `tries`    | uint32 | yes      | None              | Minimum consecutive failures for the probe to be considered failed after, having succeeded.                   | `livenessProbe.failureThreshold` **OR** `readinessProbe.failureThreshold`       |
```

Comment:
Not sure about his name. Maybe we could use the original name `failureThreshold`.
It's longer but more descriptive.